README.md

```diff
@@ -119,6 +119,166 @@ Returns the 4 byte function selector for the given function signature.
 b'\xc3x\n:'
 ```
 
+### Applicators
+
+Applicators help you apply "formatters" in various ways, most notably:
+
+- apply formatters to values by key
+- apply formatters to lists by index
+- conditionally applying a formatter
+- conditionally applying one of several formatters.
+
+Here we define a "formatter" as any `callable` that may be called with a single positional argument.
+It returns the "formatted" result. For example `int()` could be used as a formatter.
+
+Defining your own formatter is easy:
+
+```py
+def i_put_my_thing_down_flip_it_and_reverse_it(lyric):
+    return ''.join(reversed(lyric))
+```
+
+#### `apply_formatter_if(condition, formatter, value)` -> new_value
+
+This curry-able function will apply the formatter only if `bool(condition()) is True`.
```

Comment:
Should we adopt the cytoolz/toolz approach and provide an `eth_utils.curried` namespace where all of the `eth_utils` functions can be imported in their curriable format (rather than having some mis-match of curried/non-curried functions)